controller/worker/domain_migration/domain_migration.go

```diff
@@ -375,7 +375,7 @@ func (m *migration) appMaybeCreateRoute(appID string, oldRoute *router.Route, ro
 		Sticky:  oldRoute.Sticky,
 		Service: oldRoute.Service,
 	}
-	if oldRoute.Certificate != nil && oldRoute.Certificate.Cert == m.dm.OldTLSCert.Cert {
+	if oldRoute.Certificate != nil && oldRoute.Certificate.Cert == strings.TrimSuffix(m.dm.OldTLSCert.Cert, "\n") {
```

Comment:
Yeah, that would be safer. It should only ever have a trailing newline at most though.